source/tests/system/nidcpower_driver_api_tests.cpp

```diff
@@ -474,6 +486,23 @@ TEST_F(NiDCPowerDriverApiTest, ConfigureOutputFunctionAndCurrentLevel_Configures
   EXPECT_EQ(expected_output_function_value, actual_output_function_value);
 }
 
+TEST_F(NiDCPowerDriverApiTest, SetMeasureWhenAndInitiate_MeasureMultiple_ReturnsSuccess)
+{
+  const char* channel_name = ""; // all channels in session
+  // Attribute 'NIDCPOWER_ATTRIBUTE_MEASURE_WHEN' must be set to On Demand before calling MeasureMultiple
+  set_int32_attribute(
+    channel_name,
+    dcpower::NiDCPowerAttributes::NIDCPOWER_ATTRIBUTE_MEASURE_WHEN,
+    dcpower::MeasureWhen::MEASURE_WHEN_NIDCPOWER_VAL_ON_DEMAND
+  );
+  initiate();
+
+  dcpower::MeasureMultipleResponse response;
+  measure_multiple(channel_name, &response);
+
+  EXPECT_EQ(kdcpowerDriverApiSuccess, response.status());
```

Comment:
Consider also asserting on the response's `voltage_measurements` or `current_measurements` fields if that makes sense in this test context.

Reply:
I had an assert on size of `voltage_measurements` and `current_measurements`, but Jonathan had commented not to do that on similar assert on FetchMultiple API's test. Since we don't want our test to rely on which simulated hardware we would be running the test against, he mentioned that we shouldn't have assumption about size of the response.